src/app/home/home.component.ts

```diff
@@ -25,6 +32,32 @@ export class HomeComponent {
     this.page = page.pageIndex;
   }
 
+  onTabChange(tab: MatTabChangeEvent): void {
+    const index = tab.index;
+    const tabName = Object.keys(this.catalogueService.CONF.tabs)[index];
+    const path = `/tab${this.catalogueService.CONF.tabs[tabName].path}`;
+    
+    if (this.location.path() !== path) {
+      this.location.go(path);
+    }
+  }
+  
+  get isCatalogueLoading(): boolean {
+    return this.catalogueService.isLoading;
+  }
```

Comment:
is this a bit redundant? you can just get this value from the catalogue service, it's public
creating a function is not good, specially if you use this getter in the web view, this craetes a mechanism that will trigger this getter function everytime the html view is refreshed (and it's refreshed everytime you click on a button or somewhere)

If you use directly catalogueService.isLoading you will save this function call